### PR TITLE
fix(components): [date-picker] not trigger emits when type is monthrange

### LINKED ISSUE #13835: [Component] [date-picker] Datepicker did not trigger calendar-change when type is monthrange
Bug Type: **`Component`**

## Environment

- Vue Version: `3.3.2`
- Element Plus Version: `2.3.6`
- Browser / OS: `Microsoft Edge Version 113.0.1774.57 (Official build) (arm64) / macOS Monterey`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJzcmMvQXBwLnZ1ZSI6Ijx0ZW1wbGF0ZT5cbiAgPGRpdiBjbGFzcz1cImRlbW8tZGF0ZS1waWNrZXJcIj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkRlZmF1bHQ8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJtb250aHJhbmdlXCJcbiAgICAgICAgcmFuZ2Utc2VwYXJhdG9yPVwiVG9cIlxuICAgICAgICBzdGFydC1wbGFjZWhvbGRlcj1cIlN0YXJ0IG1vbnRoXCJcbiAgICAgICAgZW5kLXBsYWNlaG9sZGVyPVwiRW5kIG1vbnRoXCJcbiAgICAgICAgQGNhbGVuZGFyLWNoYW5nZT1cInVwZGF0ZURhdGVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZTEgPSByZWYoJycpXG5jb25zdCB1cGRhdGVEYXRlID0gKGRhdGU6IFtEYXRlLCBEYXRlXSkgPT4ge1xuICBjb25zb2xlLmxvZyhkYXRlKVxufVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZS1waWNrZXIge1xuICBkaXNwbGF5OiBmbGV4O1xuICB3aWR0aDogMTAwJTtcbiAgcGFkZGluZzogMDtcbiAgZmxleC13cmFwOiB3cmFwO1xufVxuLmRlbW8tZGF0ZS1waWNrZXIgLmJsb2NrIHtcbiAgcGFkZGluZzogMzBweCAwO1xuICB0ZXh0LWFsaWduOiBjZW50ZXI7XG4gIGJvcmRlci1yaWdodDogc29saWQgMXB4IHZhcigtLWVsLWJvcmRlci1jb2xvcik7XG4gIGZsZXg6IDE7XG59XG4uZGVtby1kYXRlLXBpY2tlciAuYmxvY2s6bGFzdC1jaGlsZCB7XG4gIGJvcmRlci1yaWdodDogbm9uZTtcbn1cbi5kZW1vLWRhdGUtcGlja2VyIC5kZW1vbnN0cmF0aW9uIHtcbiAgZGlzcGxheTogYmxvY2s7XG4gIGNvbG9yOiB2YXIoLS1lbC10ZXh0LWNvbG9yLXNlY29uZGFyeSk7XG4gIGZvbnQtc2l6ZTogMTRweDtcbiAgbWFyZ2luLWJvdHRvbTogMjBweDtcbn1cbjwvc3R5bGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZSxcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl1cbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJfbyI6e319)

### Steps to reproduce

1. Click the date-picker 
2. Select a range between 2 month

### What is Expected?

log 1 or 2 dates in the array at the console

### What is actually happening?

Did not trigger the function

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by listennn08**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

## Description
fix the date picker didn't trigger emits `calendar-change` when type is monthrange
closed #13835 

### <samp>🤖 Generated by Copilot at dc2bdf6</samp>

Emit `calendar-change` event from `panel-month-range.vue` to sync month range selection with parent component. This enables custom month range feature in date picker.

## Related Issue

Fixes #\_\_\_.

### <samp>🤖 Generated by Copilot at dc2bdf6</samp>

* Emit `calendar-change` event with selected month range in panel ([link](https://github.com/element-plus/element-plus/pull/13880/files?diff=unified&w=0#diff-aae21490a51f2812af5785511e78e38b1160b12da90559b82c3a6a1b9a07e308R187))

### CODE REVIEW

**Comment by xdbobname** [comment]
@HeftyKoo ,I really need it, please merge soon, THANKS VERY MUCH!

**Comment by xdbobname** [comment]
@btea could you take a look? I really need it, THANKS VERY MUCH!

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue [hunk 1]
```diff
@@ -184,6 +184,7 @@ const handleRangePick = (val: RangePickValue, close = true) => {
   if (maxDate.value === maxDate_ && minDate.value === minDate_) {
     return
   }
+  emit('calendar-change', [minDate_.toDate(), maxDate_ && maxDate_.toDate()])
```
Comment (btea) [inline packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]:
```suggestion
  emit('pick', [minDate_.toDate(), maxDate_ && maxDate_.toDate()])
```

Comment (xdbobname) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: for event 'pick', finally it turns to model value change, which conflict its original logic(skip the validate);

Comment (xdbobname) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: for 'calendar-change', I just need a event to get the intermediate picked value, it's enough and doesnt bother any other, like date-range-picker

Comment (xdbobname) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: or i misunderstood?

Comment (btea) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: You are right, I seem to be mistaken.

**Review by btea — APPROVED** [review]

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue (hunk 1 above)
Comment (tolking) [inline packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: > calendar-change | triggers when the calendar selected date is changed. Only for daterange

IMO, The description in the document may need to be updated.

**Review by ryuhangyeong — APPROVED** [review]

File: packages/components/date-picker/src/date-picker-com/panel-month-range.vue (hunk 1 above)
Comment (xdbobname) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: opps, the committer never come up, could you change the doc later?

Comment (listennn08) [reply packages/components/date-picker/src/date-picker-com/panel-month-range.vue:187]: Thanks for guys' help, I had a vacation last week, didn't notice here.

**Pull request closed without merging**